gst/nnstreamer/nnstreamer_subplugin.c

```diff
@@ -216,3 +222,47 @@ unregister_subplugin (subpluginType type, const char *name)
 
   return ret;
 }
+
+/** @brief dlclose as dealloc function for handles */
+static void
+_dlclose (gpointer data)
+{
+/**
+ * Ubuntu 16.04 / GLIBC 2.23 Workaround
+ * If we do dlclose at exit() function, it may incur
+ * https://bugzilla.redhat.com/show_bug.cgi?id=1264556#c42
+ * , which is a GLIBC bug at 2.23.
+ * The corresponding error message is:
+ * Inconsistency detected by ld.so: dl-close.c: 811:
+ * _dl_close: Assertion `map->l_init_called' failed!
+ */
+#if defined(__GLIBC__) && (__GLIBC__ == 2) && (__GLIBC_MINOR__ == 23)
+  return;                       /* Do not call dlclose and return */
```

Comment:
Then, in this case, `dlclose()` is not called to release resources?

Reply:
Calling dlclose() breaks the whole application in GLIBC 2.23 if it is called in exit(). (refer to the link in the comment). And we cannot call dlclose() before exit().

Reply:
Yes, my question was... to avoid this bug, are we going to give up to call `dlcose()` until glibc() is updated? If there is no way to prevent the memory leak in this case, it's fine.

Reply:
There is a way to prevent it: upgrade GLIBC to 2.24+ or downgrade. :)

So.. actually, we cannot avoid it as long as we cannot predict when the application (nnstreamer user) will stop using a specific subplugin.